xbmc/rendering/gl/RenderSystemGL.h

```diff
@@ -28,6 +30,50 @@ enum class ShaderMethodGL
   SM_MULTI_BLENDCOLOR
 };
 
+template<>
+struct fmt::formatter<ShaderMethodGL> : fmt::formatter<std::string>
+{
+
+public:
+  static constexpr auto toString(ShaderMethodGL method)
+  {
+    switch (method)
+    {
+      case ShaderMethodGL::SM_DEFAULT:
+        return "default";
+        break;
+      case ShaderMethodGL::SM_TEXTURE:
+        return "texture";
+        break;
+      case ShaderMethodGL::SM_TEXTURE_LIM:
+        return "texture limited";
+        break;
+      case ShaderMethodGL::SM_MULTI:
+        return "multi";
+        break;
+      case ShaderMethodGL::SM_FONTS:
+        return "fonts";
+        break;
+      case ShaderMethodGL::SM_TEXTURE_NOBLEND:
+        return "texture no blending";
+        break;
+      case ShaderMethodGL::SM_MULTI_BLENDCOLOR:
+        return "multi blend colour";
+        break;
+      default:
+        return "unknown";
+        break;
+    }
+  }
+
+  template<typename FormatContext>
+  constexpr auto format(const ShaderMethodGL& shaderMethod, FormatContext& ctx)
+  {
+    auto shaderName = toString(shaderMethod);
```

Comment:
You could have just kept it like I had it and used const std::map instead.

Reply:
Yes, The constexpr you have isn't really possible because the `std::string` (not in c++17 anyways). We should be using `std::string_view` (like I had).

The map is a much nicer 1:1 translation, however, without `constexpr` we can't do a compile time check if we catch everything. This is why I was waiting for #21084

Reply:
as already discussed, I need a fix for this unitl #21084 is ready. 
Just tell me how you want it exactly please.

Reply:
go ahead an do it like you wish. we can always change it later.

Reply:
thanks, then I'll merge it as is